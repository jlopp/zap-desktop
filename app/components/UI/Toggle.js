import React from 'react'
import styled from 'styled-components'
import { Checkbox, asField } from 'informed'

const Wrapper = styled.div`
  position: relative;

  .switch {
    position: relative;
    display: inline-block;
    width: 35px;
    height: 22px;
  }

  .switch input {
    opacity: 0;
    width: 0;
    height: 0;
  }

  .slider {
    position: absolute;
    cursor: pointer;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background-color: ${props => props.theme.colors.gray};
    transition: 0.4s;
    border-radius: 34px;
  }

  .slider:before {
    position: absolute;
    content: '';
    height: 20px;
    width: 20px;
    left: 1px;
    bottom: 1px;
    background-color: white;
    transition: 0.4s;
    border-radius: 50%;
    box-shadow: 1px 1px 3px ${props => props.theme.colors.invisibleGray};
  }

  input:checked + .slider {
    background-color: ${props => props.theme.colors.superGreen};
  }

  input:focus + .slider {
    box-shadow: 0 0 1px #2196f3;
  }

  input:checked + .slider:before {
    transform: translateX(13px);
  }
`

const Toggle = ({ fieldState, ...props }) => (
  <Wrapper>
    <label // eslint-disable-line jsx-a11y/label-has-for,jsx-a11y/label-has-associated-control
      className="switch"
    >
      <Checkbox type="checkbox" fieldState={fieldState} {...props} />
      <span className="slider round" />
    </label>
  </Wrapper>
)

export default asField(Toggle)
