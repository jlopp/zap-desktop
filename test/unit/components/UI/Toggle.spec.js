import React from 'react'
import { Form } from 'informed'
import Toggle from 'components/UI/Toggle'
import renderer from 'react-test-renderer'
import { dark } from 'themes'
import { ThemeProvider } from 'styled-components'

describe('component.UI.Toggle', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={dark}>
          <Form>
            <Toggle field="name" />
          </Form>
        </ThemeProvider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
