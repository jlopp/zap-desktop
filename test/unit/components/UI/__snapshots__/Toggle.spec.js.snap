// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component.UI.Toggle should render correctly 1`] = `
.c0 {
  position: relative;
}

.c0 .switch {
  position: relative;
  display: inline-block;
  width: 35px;
  height: 22px;
}

.c0 .switch input {
  opacity: 0;
  width: 0;
  height: 0;
}

.c0 .slider {
  position: absolute;
  cursor: pointer;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-color: #959595;
  -webkit-transition: 0.4s;
  transition: 0.4s;
  border-radius: 34px;
}

.c0 .slider:before {
  position: absolute;
  content: '';
  height: 20px;
  width: 20px;
  left: 1px;
  bottom: 1px;
  background-color: white;
  -webkit-transition: 0.4s;
  transition: 0.4s;
  border-radius: 50%;
  box-shadow: 1px 1px 3px #555;
}

.c0 input:checked + .slider {
  background-color: #39e673;
}

.c0 input:focus + .slider {
  box-shadow: 0 0 1px #2196f3;
}

.c0 input:checked + .slider:before {
  -webkit-transform: translateX(13px);
  -ms-transform: translateX(13px);
  transform: translateX(13px);
}

<form
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div
    className="c0"
  >
    <label
      className="switch"
    >
      <input
        checked={false}
        name="name"
        onBlur={[Function]}
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="slider round"
      />
    </label>
  </div>
</form>
`;
